### ISSUE TITLE
Display number of tests in summary

### ISSUE THREAD

**Issue by mkmurali** [mitre/heimdall-lite#216]
**Is your feature request related to a problem? Please describe.**
Dashboard shows summary at controls level granularity and also estimates percentage compliance level based on these counts.  However, this may not always reflect number of tests run in each control and how close the team is to achieve compliance in each control. For example, a control may be considered as failed even if 1 out of 100 tests in that control failed. 

When we ran nginx baseline @ https://github.com/mitre/nginx-baseline against the ubuntu test environment provided in the inspec training, only 13 out of 42 controls passed and 24 controls failed. If we take a look at tests level, 2,592 tests passed out of 2747. Only 148 tests failed and 7 skipped. This level of granularity in summary dashboard will provide much clearer picture for all stakeholders and reflect the progress made when used in CI/CD or review delta results.

![image](https://user-images.githubusercontent.com/763436/80768757-dd7e4900-8b10-11ea-9baa-cf2aa6944281.png)
 
Numbers by tests included (produced with a simple JSON results parser built to validate the usecase) 
![image](https://user-images.githubusercontent.com/763436/80770268-6dbe8d00-8b15-11ea-800f-d17992ce197f.png)

**Describe the solution you'd like**
1. It will be beneficial to include tests summary as well, to provide a better picture of the status of tests contributing to overall compliance. Please see the mockup, where summary of tests is also added to provide additional context.
![image](https://user-images.githubusercontent.com/763436/80770778-d823fd00-8b16-11ea-8ae6-29be4e80c4d6.png)

2. While number of tests not necessarily mean number of controls and hence compliance, it is the level of granularity that is interesting and useful to developers, testers and security engineers. it will be great to add another row (or separate tab) to display all summary by test cases (instead of summary by controls). This will be immensely helpful to developers, testers and reviewers to quickly look at ONLY failed tests for instance. 

**Describe alternatives you've considered**
Built a command line parser to view this information. Please see above for a screenshot of information produced with such a parser. 

**Desired Impact to Workflow**
Great to visualize the information in different views, as relevant to different team members that are contributing to overall compliance.

**Comment by aaronlippold** [comment]
Hi, this seems like a great improvement. 

Would you be able to attach your json parser so we could use it for reference.

**Comment by aaronlippold** [comment]
As a note - this perhaps should just be added to inspecjs formally and then just added to heimdall lite

**Comment by mkmurali** [comment]
@aaronlippold Its pretty brute force, but will help to validate the logic and results. Please see attached the parser as well as inspec results json file.
[simpleparser.zip](https://github.com/mitre/heimdall-lite/files/4588518/simpleparser.zip)

**Comment by mkmurali** [comment]
Incase if somebody is interested in using the simple parser in the meantime, I packaged the parser to npm.

```bash
npx @softrams/inspec-results-parser <inspecresults.json> [--verbose]
```

**Comment by aaronlippold** [comment]
I was just looking at this yesterday. Any thoughts on how we would want to
communicate what we're counting. Controls doesn't seem to quite make sense
and tests almost don't seem to make sense so I've been trying to figure out
what the nomenclature would be to distinguish between the control aka the
set of tests and the individual tests. Thoughts? How do you usually
communicate it when you're showing this data.

**Comment by mkmurali** [comment]
@aaronlippold Since current dashboard shows results at controls level (as x number of controls passed or failed), in my proposed screenshot I showed number of tests within those passed/failed controls and how many of them passed or failed in each group. 

I spoke to just a handful of people looking at results, most of them are security engineers and one CISO, they were able to understand and appreciate the granularity. 

For example, one take away from the example I included is that only a small fraction of tests are contributing to overall failure at control level (just 148 out of 2747 total tests are failing). More confidence in terms of where they are wrt to effort to get to the end of the tunnel.  

Based on number of downloads of that helper script on npm so far, there may be a bit of interest to see that level of information. Hope some of those users will leave some feedback/comment to take this discussion further.
